project/project.go

```diff
@@ -70,25 +79,34 @@ func (proj *Project) MergeFrom(mergeProj Project) error {
 			return nil
 		}
 
-		fileContent, err := ioutil.ReadFile(filePath)
+		fileReader, err := os.Open(filePath)
 		if err != nil {
 			return fmt.Errorf("failed to read '%s'", filePath)
 		}
 
-		if !proj.ShouldMerge(filePath, fileContent) {
+		if !proj.ShouldMerge(filePath, fileReader) {
 			return nil
 		}
 
+		fileReader.Close()
```

Comment:
You should defer closing the reader immediately after confirming there were no errors creating it. In this case if the project `ShouldMerge` returns false the function will return and the reader won't be closed leaving a dangling handler.